pyresample/geometry.py

```diff
@@ -298,6 +308,17 @@ def get_bbox_lonlats(self) -> tuple:
                            (s4_lon.squeeze(), s4_lat.squeeze())])
         if hasattr(lons[0], 'compute') and da is not None:
             lons, lats = da.compute(lons, lats)
+        # compare the first two pixels in the right column
+        lat_is_increasing = lats[1][0] < lats[1][1]
+        # compare the first two pixels in the "top" column
+        lon_is_increasing = lons[0][0] < lons[0][1]
+        is_ccw = (lon_is_increasing and lat_is_increasing) or (not lon_is_increasing and not lat_is_increasing)
```

Comment:
I can see two issues with this approach:

1. This assumes two things: a. the top and right edges of the swath are perpendicular, and b. the swath is never aligned with the equator. For b., it would indeed be a very strange orbit, so I think we can live with this assumption (although it should be mentioned at least in the code), while a. could be a bit problematic with sensors that are not our usual imagers. An alternative test could be to check that the oriented angle around the top-right corner of the swath is not reflex.
2. pole or antimeridian situations. A solution could be to convert the three points to cartesian and use the angle check described above.

Reply:
Oh good point on the poles. So I could do the math on the first corner to get the angle between the two vectors in 3D space, but won't that always give the smallest angle between them? It's been a while since I've done this type of math (and I have a cold) but looking at https://www.omnicalculator.com/math/angle-between-two-vectors, how do I know if it is clockwise or counter-clockwise in 3D space?

Reply:
because you are on the surface of a spheroid, you can give it an orientation...

Reply:
i.e. the normal of the angle has to point away from the centre of the earth

Reply:
maybe this can help? https://math.stackexchange.com/a/1715844